.github/workflows/backend-validation.yaml

```diff
@@ -0,0 +1,37 @@
+name: Validate backend application
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      working-directory: app/backend
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -r requirements-test.txt
+
+    - name: Run pytest
+      working-directory: app/backend
+      run: python -m pytest
+      env:
+        AZURE_CLIENT_ID: 00000000-0000-0000-0000-000000000000
+        AZURE_CLIENT_SECRET: Not-The-Client-Secret
+        AZURE_TENANT_ID: 00000000-0000-0000-0000-000000000000
```

Comment:
This is still failing, right? https://github.com/Azure-Samples/azure-search-openai-demo/actions/runs/5286159980/jobs/9565327057?pr=325
If this app was constructed using the app factory pattern, I think it'd be easier to pass in a mocked credential.
Are there any env variables that will get it past the Default credential stage?

Reply:
I eventually got the app factory pattern working so I could patch out `DefaultAzureCredential` during the tests.